src/utils/daos_hdlr.c

```diff
@@ -538,7 +538,16 @@ file_chmod(struct cmd_args_s *ap, struct file_dfs *file_dfs, const char *path,
 			rc = errno;
 		}
 	} else if (file_dfs->type == DAOS) {
-		rc = dfs_sys_chmod(file_dfs->dfs_sys, path, mode);
+		rc = dfs_sys_chmod(file_dfs->dfs_sys, path, mode, ignore_unsup);
+		if (rc == ENOTSUP && !ignore_unsup) {
+			/* If the file has unsupported mode bits, unset them
+			 * and retry the chmod. We track these errors so they
+			 * can be surfaced to the user at the end of the copy
+			 * operation.
+			 */
+			(*num_chmod_enotsup)++;
+			rc = dfs_sys_chmod(file_dfs->dfs_sys, path, mode, true);
+		}
```

Comment:
We could avoid an API change and the need for a new function by handling the bits here instead of within `dfs_sys_chmod`. The benefit to handling here is that we could unset the bits _**before**_ calling `dfs_sys_chmod`, avoiding the extra call.
The downside is that if other applications use `dfs_sys_chmod`, they potentially need similar logic.

@mchaarawi @ashleypittman What do you think about handling here vs in the `dfs_sys` API?